utils/src/atomics/simple_atomic.rs

```diff
@@ -0,0 +1,235 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! This module implements "simplified" atomic types, which use predefined memory orderings.
+//!
+//! They come in two variants: the "relaxed" ones, whose operations always use the "Relaxed"
+//! memory ordering, and "synchronizing" ones, whose stores are always "Release" and loads
+//! are always "Acquire".
+//!
+//! The reason for having them is that there are basically two use cases for an atomic type:
+//! 1) a low-level thread synchronization primitive;
+//! 2) a way to get interior mutability and/or circumvent borrow checker's restrictions.
+//!
+//! In the first case, the atomic operations should normally use the "synchronizing"
+//! memory orderings, such as "Acquire" and "Release", and in the second case the
+//! (generally cheaper) "Relaxed" ordering is enough.
+//!
+//! Additionally, sometimes atomic types can appear at the API boundary of a module or a package,
+//! which may lead to a situation where a load is performed in one module and a store in another.
+//! In this case, using the standard atomics effectively breaks encapsulation because,
+//! on the one hand, the memory orderings used by modules are their implementation details
+//! but, on the other hand, they must still agree with each other. Using a simplified atomic
+//! in this case basically encodes the purpose of the atomic in its type's name and makes
+//! the orderings a part of the module's interface.
+
+use crate::atomics::{atomic_traits::HasStdAtomic, AtomicNumTrait, AtomicTrait};
+use std::marker::PhantomData;
+
+use crate::concurrency_impl::sync::atomic::Ordering;
+
+// FIXME: do we also need SimpleAtomic traits or the fact that the wrapper is generic is enough?
+
+/// A predefined set of orderings to use by the [Atomic] type.
+pub trait Orderings {
+    const ORD_LOAD: Ordering;
+    const ORD_STORE: Ordering;
+    const ORD_LOAD_STORE: Ordering;
+}
```

Comment:
Do we want this trait to be open to custom impls? May consider using the sealed trait pattern to prevent that.